src/reactive/duplicity.py

```diff
@@ -40,6 +41,18 @@
 config = hookenv.config()
 
 
+def install_in_system_python(package_to_install):
+    """Install dependency in system python.
+
+    The charm use subprocess to call duplicity,
+    which is installed by apt.
+    So the azure-storage-blob has to be installed n system level python
+    """
+    subprocess.run(["sudo", "pip", "install", package_to_install])
+
+    set_flag("duplicity.azure_package_installed")
```

Comment:
**nitpick (blocking):** 

Not a nice place to run `set_flag`, this function should be more low level and only related to run subprocess.

Also error handling is required.

Reply:
👍🏼 this function name suggests nothing about `duplicity.azure_package_installed`, this line should be somewhere else

Reply:
changed this logic.